flink-connector-mongodb/src/test/java/org/apache/flink/connector/mongodb/source/splitter/MongoShardedSplitterTest.java

```diff
@@ -0,0 +1,145 @@
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.MockitoAnnotations;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.AVG_OBJ_SIZE_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.COUNT_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.DROPPED_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.ID_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.ID_HINT;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.KEY_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.MAX_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.MIN_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.SHARD_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.SIZE_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.UUID_FIELD;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mockStatic;
+
+/** Unit tests for {@link MongoShardedSplitter}. */
+@ExtendWith(TestLoggerExtension.class)
+public class MongoShardedSplitterTest {
+
+    @Mock private MongoClient mongoClient;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testShardedSplitter() {
+        MongoNamespace namespace = new MongoNamespace("test_db.test_coll");
+        BsonDocument mockCollectionMetadata = mockCollectionMetadata();
+        ArrayList<BsonDocument> mockChunksData = mockChunksData();
+
+        MongoSplitContext splitContext =
+                MongoSplitContext.of(
+                        MongoReadOptions.builder().build(),
+                        mongoClient,
+                        namespace,
+                        mockCollStats());
+
+        List<MongoScanSourceSplit> expected = new ArrayList<>();
+        for (int i = 0; i < mockChunksData.size(); i++) {
+            BsonDocument mockChunkData = mockChunksData.get(i);
+            expected.add(
+                    new MongoScanSourceSplit(
+                            String.format("%s_%d", namespace, i),
+                            namespace.getDatabaseName(),
+                            namespace.getCollectionName(),
+                            mockChunkData.getDocument(MIN_FIELD),
+                            mockChunkData.getDocument(MAX_FIELD),
+                            mockCollectionMetadata.getDocument(KEY_FIELD)));
+        }
+
+        try (MockedStatic<MongoUtils> util = mockStatic(MongoUtils.class)) {
```

Comment:
see above about mocking; this will require a bit of refactoring. Probably a sort of wrapper around the `MongoUtils`.